yt/units/yt_array.py

```diff
@@ -529,7 +529,11 @@ def __str__(self):
         """
 
         """
-        return super(YTArray, self).__str__()+' '+self.units.__str__()
+        # need to check if we're dealing with a scalar to avoid a recursion
+        # error on NumPy 1.14 and newer
+        if self.shape != ():
+            return super(YTArray, self).__str__()+' '+self.units.__str__()
+        return str(np.asarray(self)) + ' ' + str(self.units)
```

Comment:
I'd think you don't need the if-statement at all; you could just execute the second one unconditionally - or, probably a bit faster, do `str(self.view(np.ndarray)) + ' ' + str(self.units)` -- the latter is closer to what we do in `astropy`: https://github.com/astropy/astropy/blob/master/astropy/units/quantity.py#L1250